### PR TITLE
Add Sort command

### PR DESCRIPTION
**Pull request by kangqiao322**
Currently, users are unable to sort their list of internship applications by the date and time of their tasks in each internship.

Let's add a sort command to help users sort their internship applications by the date and time of their tasks.

`sort a` will sort all internships in an ascending manner, with the internship with the task with the nearest date and time at the top.

`sort d` will sort all the internships in a descending manner, with the internship with the task with the furthest date and time at the top.

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -158,6 +158,7 @@ This section describes some noteworthy details on how certain features are imple
 
 #### About this feature
 
+The add internship application feature allows users to quickly add an internship application  in the tracker via the
```
Comment (jetlfj) [inline docs/DeveloperGuide.md:161]:
```suggestion
The add internship application feature allows users to quickly add an internship application in the tracker via the
```
Changing this back

File: src/main/java/seedu/intrack/model/internship/UniqueInternshipList.java [hunk 2]
```diff
@@ -99,6 +101,8 @@ public void setInternships(List<Internship> internships) {
         internalList.setAll(internships);
     }
 
+
+
```
Comment (jetlfj) [inline src/main/java/seedu/intrack/model/internship/UniqueInternshipList.java:105]: Unnecessary white space.

**Review by jetlfj — COMMENTED** [review]
Seems to work as intended, good use of SLAP.

File: src/main/java/seedu/intrack/logic/commands/SortCommand.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        if (orderType == "a") {
+            model.ascendSort();
+            model.updateFilteredInternshipList(PREDICATE_SHOW_ALL_INTERNSHIPS);
+            return new CommandResult(String.format(MESSAGE_SUCCESS_A, model.getFilteredInternshipList().size()));
+        } else {
+            model.descendSort();
+            model.updateFilteredInternshipList(PREDICATE_SHOW_ALL_INTERNSHIPS);
+            return new CommandResult(String.format(MESSAGE_SUCCESS_D, model.getFilteredInternshipList().size()));
+        }
+    }
```
Comment (johnrhimawan) [inline src/main/java/seedu/intrack/logic/commands/SortCommand.java:43]: Perhaps we could abstract this into new classes that extends ```SortCommand``` such as classes like ```AscendSortCommand``` and ```DescendSortCommand``` to adhere to OOP?

Comment (kangqiao322) [reply src/main/java/seedu/intrack/logic/commands/SortCommand.java:43]: Due to time constraints, maybe I'll abstract this later?

File: src/main/java/seedu/intrack/model/internship/UniqueInternshipList.java (hunk 2 above)
Comment (kangqiao322) [reply src/main/java/seedu/intrack/model/internship/UniqueInternshipList.java]: Thanks for pointing that out, will be fixed.

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (kangqiao322) [reply docs/DeveloperGuide.md]: Thanks for pointing that out, will be fixed.

File: src/main/java/seedu/intrack/logic/commands/SortCommand.java (hunk 3 above)
Comment (johnrhimawan) [reply src/main/java/seedu/intrack/logic/commands/SortCommand.java:43]: Noted, that's okay!

**Pull request merged**